e2e/test/scenarios/dashboard-cards/click-behavior.cy.spec.js

```diff
@@ -202,7 +202,8 @@ describe("scenarios > dashboard > dashboard cards > click behavior", () => {
       cy.get("aside").findByText("No available targets").should("exist");
       cy.get("aside").button("Done").click();
 
-      saveDashboard();
+      // We need to wait so that the query has enough time to get resolved
+      saveDashboard({ waitMs: 250 });
```

Comment:
Do you need to wait for some query to run instead? Maybe it's possible to add a boolean flag that makes the helper wait for some queries?

Reply:
Yes, but you don't know which queries exactly. And unfortunately, cypress won't allow you to wait for all queries at once.

So if you do `cy.intercept("GET", "/api/dashboard/*").as("dashboard")` and then `cy.wait('@dashboard')`, it will "resolve" with the first successful query without waiting for the remaining ones. 

This will be address in a separate task, as [mentioned in slack](https://metaboat.slack.com/archives/C5XHN8GLW/p1703278591376449), as it is clearly outside of the scope of this bug fix.

Reply:
Could you explain why you had to increase the timeout?

Reply:
I don't have a definitive answer to this but my understanding is the following:

1. `saveDashboard` helper was throwing requests without ever waiting for them
2. the `wait(1)` inside was fixing some rendering issues, basically allowing the browser to calculate/render everything (see the comment inside the helper)
3. tests, however, were always operating on a potentially stale state/data, without waiting for **anything** after `saveDashboard` call. All of them

Number 3 is wrong. I am not sure wether I am missing something or not, but it appears that the tests were working either by accident (1 ms was enough), or stale state was matching the expectations, or actually were flaky.

Reply:
Actually, this particular in this particular test we fix additional _re-renders_ that are happening because of the `isLoading` updates. The tests that were failing were _hovering_ and trying to click the chart before it finishes rendering.

Reply:
I've tried data-fetching related fixes locally and they didn't work out. I agree with @oleggromov that increasing timeout seems to the only feasible thing here.

Reply:
`visitDashboard` helper waits for all the dashcard requests, so I guess that approach can be used for `saveDashboard` as well.